### PR TITLE
Moving the build logic into Dockerfiles + cleaning the code

### PR DESCRIPTION
**Pull request by liranmauda**
### Moving the build logic into Dockerfiles

Agent.Dockerfile
- Will build the agent without the need to supply noobaa-setup
- Adding the Git Commit hash to the package.json

builder.Dockerfile:
- Creating the noobaa/builder image that contains the dependencies for the build.

Tests.Dockerfile:
- Moved the dependencies building into builder.Dockerfile
- Moved the copy of the code infant of the build of the frontend due to dependencies

Server.Dockerfile:
- Building the server inline
- Adding the Git Commit hash to the package.json

Makefile:
- Created Makefile that builds builder, server, agent and test images
- Gets the git commit hash and passing it into the builder command

Added fix_package_json.sh for fixing the package.json if needed

### Fixed dependencies order in frontend/gulpfile.js + Package.json changes

 frontend/gulpfile.js:
- Added  'install-deps' as dependencies for 'compile-styles'

package.json:
- build:fe is not calling npm_install.js anymore because
npm install is not calling rpm run build anymore

frontend/package.json:
- Removed the condition from the prepare

### Cleaning deploy base and creating setup_platform + Cleaning the code os_utils.js:

- Removed get_yum_proxy and set_yum_proxy

system_server.js:
- Removed the call for os_utils.set_yum_proxy

server_monitor.js:
- Cleaned the code from non-docker specific code.

noobaa_init.sh:
- Removed the extract of an rpm

- Removed deploy_base.sh
- Removed add_mongo_ssl_user.js
- Creating setup_platform.sh
- Removed setup_server_swap.sh

- Fixes for removing deploy_base.sh
- .dockerignore changes
- Removed npm_install.js

### CODE REVIEW

File: src/server/bg_services/server_monitor.js [hunk 1]
```diff
@@ -75,16 +73,8 @@ async function run_monitors() {
     await _check_internal_ips();
     await _check_disk_space();
 
-    if (PLATFORM !== 'docker') {
-        await _verify_ntp_cluster_config();
-        await _verify_proxy_cluster_config();
```
Comment (nimrod-becker) [inline src/server/bg_services/server_monitor.js]: Good change, but what will we see in the FE on the server page now we don't verify these statuses? We don't have to change the FE now, but may be set to always green until we do remove it?

Comment (liranmauda) [reply src/server/bg_services/server_monitor.js]: currently on the UI, it is shown as Not configured: 
NooBaa_Management_Console

File: src/deploy/NVA_build/Tests.Dockerfile [hunk 2]
```diff
@@ -63,27 +21,34 @@ RUN echo 'PATH=$PATH:/noobaa-core/node_modules/.bin' >> ~/.bashrc
...
+# Setting GIT_COMMIT for the base
+# In order to set it we need to run build with 
+# --build-arg GIT_COMMIT=$(git rev-parse HEAD)
 ##############################################################
-COPY . ./
+COPY ./package*.json ./
+ARG GIT_COMMIT 
+RUN if [ "${GIT_COMMIT}" != "" ]; then sed -i 's/^  "version": "\(.*\)",$/  "version": "\1-'${GIT_COMMIT:0:7}'",/' package.json; fi
 
 FROM base as unitest
```
Comment (dannyzaken) [inline src/deploy/NVA_build/Tests.Dockerfile]: not necessarily unit tests. can be for test_env_builder too, isn't it?

File: src/deploy/NVA_build/Tests.Dockerfile [hunk 3]
```diff
@@ -94,9 +59,8 @@ FROM base as unitest
 #   Cache: rebuild when we adding/removing requirments
 ##############################################################
 ENV TEST_CONTAINER true
-RUN yum install -y -q centos-release-scl && \
-    yum install -y -q rh-mongodb36 && \
-    yum install -y ntpdate && \ 
+RUN yum install -y -q rh-mongodb36 && \
```
Comment (dannyzaken) [inline src/deploy/NVA_build/Tests.Dockerfile]: `kubectl` is required to run test_env_builder

**Review by nimrod-becker — APPROVED** [review]

**Pull request merged**